scripts/crc-deploy.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+
+set -e
+
+# This script manages the deployment of downstream olm on a local CRC cluster
+# It will deploy the local images: olm:test and opm:test
+# Both built with make build/olm-container and build/registry-container respectively
+
+echo "Deploying OLM to CRC"
+
+# Set kubeconfig to CRC if necessary
+export KUBECONFIG=${KUBECONFIG:-${HOME}/.crc/machines/crc/kubeconfig}
+KUBE_ADMIN_PASSWORD=$(tr -d '\n' < "${HOME}/.crc/machines/crc/kubeadmin-password")
+
+# CRC registry coordinates
+OPENSHIFT_REGISTRY_NAMESPACE="openshift-image-registry"
+OLM_NAMESPACE="openshift-operator-lifecycle-manager"
+IMAGE_REGISTRY_SVC="image-registry"
+IMAGE_REGISTRY_PORT=5000
+
+# login to crc
+echo "Logging in as kubeadmin"
+oc login -u kubeadmin -p "${KUBE_ADMIN_PASSWORD}" > /dev/null
+
+# Scale down CVO to stop changes from returning to stock configuration
+echo "Scaling down CVO"
+oc scale --replicas 0 -n openshift-cluster-version deployments/cluster-version-operator
+
+SKIP_MANIFESTS=${SKIP_MANIFESTS:-0}
+if [ "${SKIP_MANIFESTS}" = 0 ]; then
+  # Build e2e manifests
+  echo "Generating manifests"
+  CRC_E2E=1 ./scripts/generate_crds_manifests.sh
+
+  # Use the numbered ordering in the manifest file names to delete and deploy the manifests in order
+  # First in reverse with tac to delete then in order to recreate
+  echo "Deploying OLM"
+  find_flags=(-regex ".*\.yaml" -not -regex ".*\.removed\.yaml" -not -regex ".*.ibm-cloud-managed\.yaml")
+  echo "Deleting manifests"
+  find manifests "${find_flags[@]}" | tac | while read -r manifest; do
+    echo "Deleting ${manifest}"
+    set +e
+    kubectl delete --ignore-not-found=true -f "${manifest}"
+    set -e
+  done
+
+  echo "Applying manifests"
+  find manifests "${find_flags[@]}" | while read -r manifest; do
+    echo "Applying ${manifest}"
+    set +e
+    kubectl create -f "${manifest}"
+    set -e
+  done
+fi
+
+# Create port-forward to CRC registry
+kubectl port-forward -n ${OPENSHIFT_REGISTRY_NAMESPACE} svc/${IMAGE_REGISTRY_SVC} ${IMAGE_REGISTRY_PORT}:${IMAGE_REGISTRY_PORT} > /dev/null 2>&1 &
+PORT_FWD_PID=$!
+
+# Remember to close the port-forward
+trap 'kill "${PORT_FWD_PID}"' EXIT
+
+# give port-forward a second
+# I found that without this docker login would not work
+# as if it couldn't reach the docker server
+sleep 1
+
+oc whoami -t | docker login localhost:5000 --username user --password-stdin
+
+# Push olm image
+echo "Pushing olm image"
+docker tag olm:test localhost:5000/openshift/olm:test
+docker push localhost:5000/openshift/olm:test
+
+# Push registry image
+echo "Pushing registry image"
+docker tag opm:test localhost:5000/openshift/opm:test
+docker push localhost:5000/openshift/opm:test
```

Comment:
Maybe this can be thrown in a function if we're trapping signals and require cleanup?

Reply:
Also variable to track any of the opm/olm test images?

Reply:
Fixed